examples/chef/common/clusters/channel/ChannelManager.cpp

```diff
@@ -0,0 +1,351 @@
+        mCurrentChannel      = mChannels[iMatchedChannel];
+        mCurrentChannelIndex = iMatchedChannel;
+        helper.Success(response);
+    }
+}
+
+bool ChannelManager::HandleChangeChannelByNumber(const uint16_t & majorNumber, const uint16_t & minorNumber)
+{
+    uint16_t index = 0;
+    for (auto const & channel : mChannels)
+    {
+        // verify if major & minor matches one of the channel from the list
+        if (channel.minorNumber == minorNumber && channel.majorNumber == majorNumber)
+        {
+            // verify if channel changed by comparing values of current channel with the requested channel
+            if (channel.minorNumber != mCurrentChannel.minorNumber || channel.majorNumber != mCurrentChannel.majorNumber)
+            {
+                mCurrentChannelIndex = index;
+                mCurrentChannel      = channel;
+                return true;
+            }
+        }
+        index++;
+    }
+    return false;
+}
+
+bool ChannelManager::HandleSkipChannel(const int16_t & count)
+{
+    int32_t newChannelIndex = static_cast<int32_t>(count) + static_cast<int32_t>(mCurrentChannelIndex);
+    uint16_t channelsSize   = static_cast<uint16_t>(mChannels.size());
+
+    // handle newChannelIndex out of range.
+    newChannelIndex = newChannelIndex % channelsSize;
+
+    if (newChannelIndex < 0)
+    {
+        newChannelIndex = newChannelIndex + channelsSize;
+    }
+
+    mCurrentChannelIndex = static_cast<uint16_t>(newChannelIndex);
+    mCurrentChannel      = mChannels[mCurrentChannelIndex];
+    return true;
+}
+
+void ChannelManager::HandleGetProgramGuide(
+    CommandResponseHelper<ProgramGuideResponseType> & helper, const chip::Optional<uint32_t> & startTime,
+    const chip::Optional<uint32_t> & endTime,
+    const chip::Optional<chip::app::DataModel::DecodableList<ChannelInfoType>> & channelList,
+    const chip::Optional<PageTokenType> & pageToken, const chip::Optional<chip::BitMask<RecordingFlagBitmap>> & recordingFlag,
+    const chip::Optional<chip::app::DataModel::DecodableList<AdditionalInfoType>> & externalIdList,
+    const chip::Optional<chip::ByteSpan> & data)
+{
+
+    // 1. Decode received parameters
+    // 2. Perform search
+    // 3. Return results
+
+    // PageTokenType paging;
+    // paging.limit  = MakeOptional(static_cast<uint16_t>(10));
+    // paging.after  = MakeOptional(chip::CharSpan::fromCharString("after-token"));
+    // paging.before = MakeOptional(chip::CharSpan::fromCharString("before-token"));
+
+    // ChannelPagingStructType channelPaging;
+    // channelPaging.nextToken = MakeOptional<DataModel::Nullable<Structs::PageTokenStruct::Type>>(paging);
+
+    std::vector<ProgramType> matches;
+    for (auto const & program : mPrograms)
+    {
+        if (startTime.ValueOr(0) > program.startTime)
+        {
+            continue;
+        }
+        if (endTime.ValueOr(0) < program.endTime)
```

Comment:
if there is no end time, 0 seems to always match. Should this be some ::MAX value instead?

Reply:
OK. Use `std::numeric_limits<uint32_t>::max()` as default endTime